Sources/PostgresNIO/New/Data/Float+PSQLCodable.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  File.swift
+//  
+//
+//  Created by Fabian Fett on 12.01.21.
+//
+
+extension Float: PSQLCodable {
+    var psqlType: PSQLDataType {
+        .float4
+    }
+    
+    static func decode(from buffer: inout ByteBuffer, type: PSQLDataType, context: PSQLDecodingContext) throws -> Float {
+        switch type {
+        case .float4:
+            guard buffer.readableBytes == 4, let float = buffer.readFloat() else {
```

Comment:
Checking `readableBytes` first is a waste of time, `ByteBuffer` does that check for you (it's one of the reasons the read methods can return `nil`).

Reply:
the read bytes check if enough bytes are available, but they don't check if there are more than the needed number of bytes. Fair point though in those cases we should probably create a PSQLBackendMessage.DecodingError since this is obviously an error in the received message. wdyt?

Reply:
Oh, yes, I missed that you were doing an exact equality check. Still, I'd prefer to spell that more like this:
```swift
extension ByteBuffer {
    internal var isAtEnd: Bool {
        return self.readableBytes == 0 // readableBytes accounts for the current readerIndex
    }
}
```
```suggestion
            guard let float = buffer.readFloat(), buffer.isAtEnd else {
```
This way the number of bytes in the type is not being hardcoded in more places than necessary.

I agree with throwing a `DecodingError` for a violation as well.